File: nhglib-core/src/io/github/voidzombie/nhglib/data/models/serialization/physics/shapes/CapsuleShapeJson.java
package io.github.voidzombie.nhglib.data.models.serialization.physics.shapes;

import com.badlogic.gdx.utils.JsonValue;
import io.github.voidzombie.nhglib.physics.models.CapsuleRigidBodyShape;

/**
 * Created by Fausto Napoli on 16/05/2017.
 */
public class CapsuleShapeJson extends ShapeJson {
    private CapsuleRigidBodyShape shape;

    @Override
    public void parse(JsonValue jsonValue) {
        float radius = jsonValue.getFloat("radius", 1.0f);
        float height = jsonValue.getFloat("height", 1.0f);

        shape = new CapsuleRigidBodyShape(radius, height);
    }

    @Override
    public CapsuleRigidBodyShape get() {
        return shape;
    }
}

File: nhglib-core/src/io/github/movementspeed/nhglib/data/models/serialization/physics/shapes/CapsuleShapeJson.kt
package io.github.movementspeed.nhglib.data.models.serialization.physics.shapes

import com.badlogic.gdx.utils.JsonValue
import io.github.movementspeed.nhglib.physics.models.CapsuleRigidBodyShape

/**
 * Created by Fausto Napoli on 16/05/2017.
 */
class CapsuleShapeJson : ShapeJson() {
    private var shape: CapsuleRigidBodyShape? = null

    override fun parse(jsonValue: JsonValue) {
        val radius = jsonValue.getFloat("radius", 1.0f)
        val height = jsonValue.getFloat("height", 1.0f)

        shape = CapsuleRigidBodyShape(radius, height)
    }

    override fun get(): CapsuleRigidBodyShape? {
        return shape
    }
}
